File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    if n == 1:
        return 1
    if n == 2:
        return 2
    jump_bound: list[int] = [0] * (n+1)
    jump_bound[1] = 1
    jump_bound[2] = 2

    for i in range(3, n + 1):
        jump_bound[i] = jump_bound[i-1] + jump_bound[i-2]
    
    return jump_bound[n] % 1234567

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        long answer = 0;
        if (n == 1) {
            return 1;
        }
        if (n == 2) {
            return 2;
        }
        
        int[] jump = new int[n+1];
        jump[1] = 1;
        jump[2] = 2;
        
        for (int i = 3; i <= n; i++) {
            jump[i] = (jump[i-1] + jump[i-2]) % 1234567;
        }
        
        return jump[n];
    }
}